cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -272,21 +316,68 @@ type itemToPurge struct {
 	purge        func(skipPurge bool) error
 }
 
+func (p *BicepProvider) scopeForTemplate(ctx context.Context, t azure.ArmTemplate) (infra.Scope, error) {
```

Comment:
If you have time to do so (can be a follow, if we've done due dilligence with the manual testing), let's do add an end-to-end that covers provisioning a storage account inside a resource group, and verify both `up` and `down` behaviors.

Reply:
Added a simple test here.